### PR TITLE
ui-spacetimechart: add ConflictLayer

### LINKED ISSUE #416: spacetimechart: add conflicts layer
Add a basic conflicts layer, with a checkbox to enable the layer. The border is out-of-scope.

### PR DESCRIPTION
**Pull request by emersion**
Super basic for now. Will be enhanced later on with borders and a tooltip.

Closes: https://github.com/OpenRailAssociation/osrd-ui/issues/416

### CODE REVIEW

File: ui-spacetimechart/src/stories/layers.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+import type { Meta, StoryObj } from '@storybook/react';
+
+import { OPERATIONAL_POINTS, PATHS, START_DATE } from './lib/paths';
+import { X_ZOOM_LEVEL, Y_ZOOM_LEVEL } from './lib/utils';
+import { SpaceTimeChart, PathLayer } from '../';
```
Comment (SharglutDev) [inline ui-spacetimechart/src/stories/layers.stories.tsx:7]: I have a linter error asking to move this line before l.5 (in all stories actually), strange that it hasn't been catched here or that you don't have it.

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested, good job ! I would have some questions, but as it's only the first step of the layer, the rest gonna probably come on other PR so I'll check for that on the next ones :)

File: ui-spacetimechart/src/stories/layers.stories.tsx (hunk 1 above)
Comment (emersion) [reply ui-spacetimechart/src/stories/layers.stories.tsx:7]: Weird. Seems like CI is fine with the current ordering as well.

**Review by RomainValls — APPROVED** [review]
LGTM and tested ✅

**Pull request merged**